tests/integration/components/denali-select-test.js

```diff
@@ -0,0 +1,145 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { render, setupOnerror, find, fillIn } from '@ember/test-helpers';
+import { hbs } from 'ember-cli-htmlbars';
+
+const option = { text: 'Item 1' };
+
+module('Integration | Component | denali-select', function (hooks) {
+  setupRenderingTest(hooks);
+
+  test('it requires options', async function (assert) {
+    assert.expect(1);
+
+    setupOnerror(function ({ message }) {
+      assert.equal(
+        message,
+        'Failed prop type: The prop `options` is marked as required in `DenaliSelectComponent`, but its value is `undefined`.',
+        'If @args() contains a PropType validator, an error will be thrown if the value is incorrect'
+      );
+    });
+    await render(hbs`<DenaliSelect />`);
+  });
+
+  test('it renders an option string', async function (assert) {
+    assert.expect(4);
+
+    await render(hbs`
+      <DenaliSelect @options={{array "Item 1"}} @onChange={{this.onChange}} as |option|>
+        {{option}}
+      </DenaliSelect>
+    `);
+
+    assert.dom('div').hasClass('input', 'DenaliSelect renders a select input.');
+
+    assert.dom('div').hasClass('has-arrow', 'DenaliSelect renders an arrow.');
+
+    assert.dom('select').exists('DenaliSelect renders a select element.');
+
+    assert.dom('option').hasText('Item 1', 'DenaliSelect option exists and renders text.');
+  });
+
+  test('it renders an option object', async function (assert) {
+    assert.expect(4);
+
+    await render(hbs`
+      <DenaliSelect @options={{array (hash text="Item 1")}} as |option|>
+        {{option.text}}
+      </DenaliSelect>
+    `);
+
+    assert.dom('div').hasClass('input', 'DenaliSelect renders a select input.');
+
+    assert.dom('div').hasClass('has-arrow', 'DenaliSelect renders an arrow.');
+
+    assert.dom('select').exists('DenaliSelect renders a select element.');
+
+    assert.dom('option').hasText('Item 1', 'DenaliSelect option exists and renders an object.');
+  });
+
```

Comment:
```suggestion
```

Do we need both string an object asserts? You _could_ pass an array of arrays of objects....

Plus we have objects in the test below